src/main/java/com/salesforce/dataloader/action/visitor/BulkV2Connection.java

```
@@ -0,0 +1,364 @@
+                        handler.handleResponse(url, bytes);
+                    }
+                }
+            }
+
+            if (getConfig().isTraceMessage()) {
+                getConfig().getTraceStream().println(url.toExternalForm());
+
+                Map<String, List<String>> headers = connection.getHeaderFields();
+                for (Map.Entry<String, List<String>>entry : headers.entrySet()) {
+                    StringBuffer sb = new StringBuffer();
+                    List<String> values = entry.getValue();
+
+                    if (values != null) {
+                        for (String v : values) {
+                            sb.append(v);
+                        }
+                    }
+
+                    getConfig().getTraceStream().println(entry.getKey() + ": " + sb.toString());
+                }
+
+                getConfig().teeInputStream(bytes);
+            }
+        }
+
+        if (!success) {
+            ContentType type = null;
+            String contentTypeHeader = connection.getContentType();
+            if (contentTypeHeader != null) {
+                if (contentTypeHeader.contains(XML_CONTENT_TYPE)) {
+                    type = ContentType.XML;
+                } else if (contentTypeHeader.contains(JSON_CONTENT_TYPE)) {
+                    type = ContentType.JSON;
+                }
+            }
+            parseAndThrowException(in, type);
+        }
+
+        this.queryLocator = connection.getHeaderField("Sforce-Locator");
+        this.queryResultNumberOfRecords = connection.getHeaderField("Sforce-NumberOfRecords");
+        return in;
+    }
+    
+    public String getQueryLocator() {
+        return this.queryLocator;
+    }
+
+    public JobInfo closeJob(String jobId) throws AsyncApiException {
+        
+        JobInfo job = new JobInfo();
+        job.setId(jobId);
+        job.setState(JobStateEnum.Closed);
+        return job;
+        /*
+        return updateJob(job);
+        */
```

There is no "close" operation for bulk v2 jobs. So maybe this method should throw UnsupportedOperationException.

---

Agreed. I will skip closing a job for bulk v2 query and remove the method from BulkV2Connection.java.

---

Good point. I have removed closeJob() from BulkV2Connection.